### ISSUE TITLE
[TextInputLayout] Add support for `activated` state

### ISSUE THREAD

**Issue by sabinebarrera** [material-components/material-components-android#2092]
**Is your feature request related to a problem? 
Material design shows that `Filled/Outlined text fields` states are:
* Inactive
* Focused
* **Activated**
* Hover
* Error
* Disabled

**Describe the solution you'd like**
There is no **activated** state:
```
<selector xmlns:android="http://schemas.android.com/apk/res/android">
  <item android:alpha="0.16" android:color="?attr/colorOnSurface" android:state_hovered="true"/>
  <item android:alpha="0.12" android:color="?attr/colorOnSurface" android:state_focused="true"/>
  <item android:alpha="0.04" android:color="?attr/colorOnSurface" android:state_enabled="false"/>
  <item android:alpha="0.12" android:color="?attr/colorOnSurface"/>
</selector>
```

**Additional context**
When the user taps on a `date text field`, the app will show the picker, therefore showing this `date text field` in an active state it is important b/c it provides context to the user. See design section below.

<details>
<summary>Design Section</summary>

<img src="https://user-images.githubusercontent.com/1494577/108787395-7eb80f00-752a-11eb-9cd7-91cc949a1cb5.png" width="600"/>
</details>

**Comment by ornolisvt** [comment]
Yes, please add the possibility of adding new drawable states and setting the corresponding stroke color. TextInputLayout is not open for extension violating well known programming principles. I want to add a new state that defines when a text field is filled and it is not possible
